crates/protogen/src/sqlexec/copy_to.rs

```diff
@@ -90,12 +90,18 @@ pub struct CopyToFormatOptionsCsv {
 pub struct CopyToFormatOptionsJson {
     #[prost(bool, tag = "1")]
     pub array: bool,
+
+    #[prost(string, repeated, tag = "2")]
+    pub partition_columns: Vec<String>,
```

Comment:
I think we need to do this for CSV, Lance, and BSON. 

I realized that we missed having one of these options structs for bson which I added in #2704

Reply:
I merged in #2704 which should make it easier to add partitioning support there too.